src/gsy_e/gsy_e_core/market_counters.py

```diff
@@ -0,0 +1,113 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from gsy_framework.constants_limits import ConstSettings
+from pendulum import DateTime
+
+
+class MarketCounter:
+    """Base class for market counters"""
+
+    def __init__(self, clearing_interval_min: int):
+        self._last_clearing_time = None
+        self.clearing_interval = clearing_interval_min
+
+    def is_time_for_clearing(self, current_time: DateTime) -> bool:
+        """Return if it is time for clearing according to self.clearing_interval."""
```

Comment:
> Good that you kept the default implementation here for the existing markets! I think that we will revisit this approach in the future since the cron-based approach that you used for the other markets is much more easy to reason about, and it probably makes sense to make it the default behavior for the existing markets too. I would not do it now though, but at a later stage, in order to resolve any issues that might pop up with the new approach. I opened already an enhancement ticket that deals with this:
> https://gridsingularity.atlassian.net/browse/GSYE-342
> and leave it in the backlog for now.

Thanks! This will also make https://gridsingularity.atlassian.net/browse/GSYE-322 possible. I will linkt the two tickets if you do not mind.

> One more remark, we could also use this approach for the update_frequency module, in order to solve the bugs that we have encountered in the price update of the strategies by using this approach. IMO it will lead to far less uncertainties and a more robust and performant algorithm than the tick-counting approach that we use now.

👍  I will also link this ticket: https://gridsingularity.atlassian.net/browse/GSYE-332